autora/cycle/plot_utils.py

```diff
@@ -0,0 +1,470 @@
+    Args:
+        cycle: AER Cycle object that has been run
+        steps: Number of steps to define the condition space
+
+    Returns: np.ndarray
+
+    """
+    l_min_max = min_max_observations(cycle)
+    l_space = []
+
+    for min_max in l_min_max:
+        l_space.append(np.linspace(min_max[0], min_max[1], steps))
+
+    return np.meshgrid(*l_space)
+
+
+def theory_predict(cycle: Cycle, conditions: Sequence) -> dict:
+    """
+    Gets theory predictions over conditions space and saves results of each cycle to a dictionary.
+    Args:
+        cycle: AER Cycle object that has been run
+        conditions: Condition space. Should be an array of grouped conditions.
+
+    Returns: dict
+
+    """
+    d_predictions = {}
+    for i, theory in enumerate(cycle.data.theories):
+        d_predictions[i] = theory.predict(conditions)
+
+    return d_predictions
+
+
+def check_replace_default_kw(default: dict, user: dict) -> dict:
+    """
+    Combines the key/value pairs of two dictionaries, a default and user dictionary. Unique pairs
+    are selected and user pairs take precedent over default pairs if matching keywords. Also works
+    with nested dictionaries.
+
+    Returns: dict
+    """
+    # Copy dict 1 to return dict
+    d_return = default.copy()
+    # Loop by keys in dict 2
+    for key in user.keys():
+        # If not in dict 1 add to the return dict
+        if key not in default.keys():
+            d_return.update({key: user[key]})
+        else:
+            # If value is a dict, recurse to check nested dict
+            if isinstance(user[key], dict):
+                d_return.update(
+                    {key: check_replace_default_kw(default[key], user[key])}
+                )
+            # If not a dict update the default value with the value from dict 2
+            else:
+                d_return.update({key: user[key]})
+
+    return d_return
+
+
+def plot_results_panel_2d(
+    cycle: Cycle,
+    iv_name: Optional[str] = None,
+    dv_name: Optional[str] = None,
+    steps: int = 50,
+    wrap: int = 4,
+    spines: bool = False,
+    subplot_kw: dict = {},
+    scatter_kw1: dict = {},
+    scatter_kw2: dict = {},
+    line_kw: dict = {},
+) -> plt.figure:
+    """
+    Generates a multi-panel plot with each panel showing results of an AER cycle. Observed data
+    is plotted as a scatter plot with the current cycle colored differently than observed data from
+    previous cycles. The current cycle's theory is plotted as a line over the range of the observed
+    data.
+    Args:
+        cycle: AER Cycle object that has been run
+        iv_name: Independent variable name. Name should match the name instantiated in the cycle
+                    object. Default will select the first.
+        dv_name: Single dependent variable name. Name should match the names instantiated in the
+                    cycle object. Default will select the first DV.
```

Comment:
I suppose it could go either way. In terms of user experience the difference would be what knowledge they have of the cycle initialization. 

1. Index requires the user remember of the order variables of the metadata. 
2. Name requires the user to know the label of the variables.

What do you think is easier?

Reply:
I think the name is easier because people will have that in mind when plotting... But not sure?

Reply:
We'll keep name for now.